djgentelella/widgets/files.py

```diff
@@ -22,17 +26,39 @@ def __init__(self, attrs=None, extraskwargs=True):
 
     def format_value(self, value):
         """File input never renders a value."""
-        return value
+        if value:
+            name = Path(value.name).name
+            self.value = mark_safe(
+                '{"name": "%(name)s", "display_name": "%(display_name)s", "url": "%(url)s" }' % {
+                    'name': value.name,
+                    'display_name': name,
+                    'url': value.url})
+            return self.value
+        return ''
+
+    def parse_value(self, value):
+        dev = None
+        try:
+            dev = json.loads(value)
+            if not ('url' in dev or 'token' in dev or 'actions' in dev):
+                dev = None
+        except Exception as e:
+            pass
+        return dev
```

Comment:
We should try to avoid broad exceptions `Exception` this is well-know as an anti-pattern and can be solved just limiting the exception to only the exception that `json.loads` is supposed to raise. https://realpython.com/the-most-diabolical-python-antipattern/

Also instead of pass we can report this to the `logger` using `logger.exception` in that way The logger will trace this error for further analysis.

Reply:
Sure, thanks, I pushed correction for this.